pkg/kubelet/container/runtime.go

```diff
@@ -57,7 +57,10 @@ type Runtime interface {
 	Version() (Version, error)
 	// APIVersion returns the API version information of the container
 	// runtime. This may be different from the runtime engine's version.
+	// TODO(random-liu): We should fold this into Version()
 	APIVersion() (Version, error)
+	// CheckStatus checks whether the underlying runtime is working normally.
+	CheckStatus() error
```

Comment:
By calling it `Status`, we can return more useful information in the
future. For example, we can return a Status object that has some metrics,
or internal state, etc. Feel free to push back :) Healthy arguments are
very helpful to maintain a project :)

On Fri, Mar 4, 2016 at 4:39 PM, Lantao Liu <notifications@github.com> wrote:

> In pkg/kubelet/container/runtime.go
> <https://github.com/kubernetes/kubernetes/pull/22467#discussion_r55109379>
> :
>
> >  	APIVersion() (Version, error)
> > +	// CheckStatus checks whether the underlying runtime is working normally.
> > +	CheckStatus() error
>
> If it is called Status(), I feel like it will return some kind of
> "status", but it only returns error or not. While, if it is called
> CheckStatus(), I could understand that it is trying to check the status,
> and may encounter some error during checking.
> I'm not good at naming, just MHO. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22467/files#r55109379>.
>